Library/Homebrew/download_strategy.rb

```diff
@@ -282,6 +283,14 @@ def initialize(name, resource)
   end
 
   def fetch
+    if ENV["HOMEBREW_ARTIFACT_DOMAIN"]
+      if @url.start_with?(BottleSpecification::DEFAULT_DOMAIN)
+        @url.sub!(BottleSpecification::DEFAULT_DOMAIN, ENV["HOMEBREW_ARTIFACT_DOMAIN"])
```

Comment:
Will making the bottle artifact url structure the same as non-bottle one cause any troubles? If not, I would prefer a consistent and unanimous way to handle artifact. Presuming you fetching bottle resources the same way as non-bottle ones, I think they are equally easy.

The more important issue here is that if we want to add support for artifact, let's make it more generalize to all the users instead of specific to your settings. That means other users can use the same function without setting bottles fetch policy differently for bottles even when they are using services other than JForg Aritfactory.